File: backend/src/main/java/com/smartcampost/backend/service/ScanEventService.java
package com.smartcampost.backend.service;

import com.smartcampost.backend.dto.scan.ScanEventCreateRequest;
import com.smartcampost.backend.dto.scan.ScanEventResponse;
import com.smartcampost.backend.dto.scan.OfflineSyncRequest;
import com.smartcampost.backend.dto.scan.OfflineSyncResponse;

import java.util.List;
import java.util.UUID;

public interface ScanEventService {

    // US38: record a scan event (with mandatory GPS)
    ScanEventResponse recordScanEvent(ScanEventCreateRequest request);

    // US39: full history for a parcel
    List<ScanEventResponse> getHistoryForParcel(UUID parcelId);

    // Offline sync: process batch of queued events
    OfflineSyncResponse syncOfflineEvents(OfflineSyncRequest request);

    // Get last scan event for parcel (current location)
    ScanEventResponse getLastScanEvent(UUID parcelId);
}

File: smartcampost-frontend/src/services/scanEventService.ts
import api from "./api";
import type { ScanEventCreateRequest, ScanEventResponse } from "../types/Scan";

export async function recordScanEvent(payload: ScanEventCreateRequest) {
  const { data } = await api.post<ScanEventResponse>("/scan-events", payload);
  return data;
}

export async function getParcelScanHistory(parcelId: string) {
  const { data } = await api.get<ScanEventResponse[]>(
    `/scan-events/parcel/${parcelId}`
  );
  return data;
}
